storybook/pages/CreateChannelPopupPage.qml

```diff
@@ -19,15 +19,105 @@ SplitView {
 
         PopupBackground {
             anchors.fill: parent
+
+            Button {
+                anchors.centerIn: parent
+                text: "Reopen"
+
+                onClicked: dialog.open()
+            }
         }
 
         CreateChannelPopup {
+            id: dialog
             anchors.centerIn: parent
             modal: false
             closePolicy: Popup.NoAutoClose
 
             isEdit: isEditCheckBox.checked
             isDeleteable: isDeleteableCheckBox.checked
+            isDiscordImport: isDiscordCheckBox.checked
+
+            Binding on channelName {
+                value: "test-channel"
+                when: dialog.isEdit
+            }
+
+            Binding on channelDescription {
+                value: "TEST TEST TEST"
+                when: dialog.isEdit
+            }
+
+            Binding on channelColor {
+                value: "pink"
+                when: dialog.isEdit
+            }
+
+            communitiesStore: QtObject {
+                property string discordImportChannelName
+                readonly property bool discordImportInProgress: false
+                readonly property bool discordDataExtractionInProgress: false
+                readonly property int discordImportErrorsCount: 0
+                readonly property int discordImportWarningsCount: 0
+                readonly property int discordOldestMessageTimestamp: 0
+
+                property var discordFileList: ListModel {
+                    readonly property int selectedCount: count
+                    property bool selectedFilesValid
+                }
+
+                property var discordCategoriesModel: ListModel {}
+
+                property var discordChannelsModel: ListModel {
+                    property bool hasSelectedItems
+                    readonly property int count: 32 // hide the parsing/loading spinner
+                }
+
+                function setFileListItems(filePaths) {
+                    for (const filePath of filePaths) {
+                        discordFileList.append({"filePath": filePath, errorMessage: ""})
+                    }
+                }
+
+                function removeFileListItem(path) {
+                    for (var i = 0; i < discordFileList.count; i++) {
```

Comment:
`var` -> `let`, basically in most cases there is no reason to use `var` in modern js. block-scoped `let`/`const` should be used instead